src/vmm/src/resources.rs

```diff
@@ -539,4 +653,240 @@ mod tests {
+            ht_enabled: Some(true),
+            cpu_template: Some(CpuFeaturesTemplate::T2),
+        };
+
+        assert_ne!(vm_resources.vm_config, aux_vm_config);
+        vm_resources.set_vm_config(&aux_vm_config).unwrap();
+        assert_eq!(vm_resources.vm_config, aux_vm_config);
+
+        // Invalid vcpu count
+        aux_vm_config.vcpu_count = Some(0);
+        assert_eq!(
+            vm_resources.set_vm_config(&aux_vm_config),
+            Err(VmConfigError::InvalidVcpuCount)
+        );
+        aux_vm_config.vcpu_count = Some(33);
+        assert_eq!(
+            vm_resources.set_vm_config(&aux_vm_config),
+            Err(VmConfigError::InvalidVcpuCount)
+        );
+        aux_vm_config.vcpu_count = Some(32);
+
+        // Invalid mem_size_mib
+        aux_vm_config.mem_size_mib = Some(0);
+        assert_eq!(
+            vm_resources.set_vm_config(&aux_vm_config),
+            Err(VmConfigError::InvalidMemorySize)
+        );
+    }
+
+    #[test]
+    fn test_boot_config() {
+        let vm_resources = default_vm_resources();
+        let expected_boot_cfg = vm_resources.boot_config.as_ref().unwrap();
+        let actual_boot_cfg = vm_resources.boot_source().unwrap();
+
+        assert_eq!(actual_boot_cfg, expected_boot_cfg);
+    }
+
+    #[test]
+    fn test_set_boot_source() {
+        let tmp_file = TempFile::new().unwrap();
+        let cmdline = "reboot=k panic=1 pci=off nomodules 8250.nr_uarts=0";
+        let expected_boot_cfg = BootSourceConfig {
+            kernel_image_path: String::from(tmp_file.as_path().to_str().unwrap()),
+            initrd_path: Some(String::from(tmp_file.as_path().to_str().unwrap())),
+            boot_args: Some(cmdline.to_string()),
+        };
+
+        let mut vm_resources = default_vm_resources();
+        let boot_cfg = vm_resources.boot_source().unwrap();
+        let tmp_ino = tmp_file.as_file().metadata().unwrap().st_ino();
+
+        assert_ne!(boot_cfg.cmdline.as_str(), cmdline);
+        assert_ne!(boot_cfg.kernel_file.metadata().unwrap().st_ino(), tmp_ino);
+        assert_ne!(
+            boot_cfg
+                .initrd_file
+                .as_ref()
+                .unwrap()
+                .metadata()
+                .unwrap()
+                .st_ino(),
+            tmp_ino
+        );
+
+        vm_resources.set_boot_source(expected_boot_cfg).unwrap();
+        let boot_cfg = vm_resources.boot_source().unwrap();
+        assert_eq!(boot_cfg.cmdline.as_str(), cmdline);
+        assert_eq!(boot_cfg.kernel_file.metadata().unwrap().st_ino(), tmp_ino);
+        assert_eq!(
+            boot_cfg
+                .initrd_file
+                .as_ref()
+                .unwrap()
+                .metadata()
+                .unwrap()
+                .st_ino(),
+            tmp_ino
+        );
+    }
+
+    #[test]
+    fn test_set_block_device() {
+        let mut vm_resources = default_vm_resources();
+
+        // Use the already existing block device config
```

Comment:
This comment is not very helpful. I would say something like: `Clone the existing block device in order to obtain a new one without creating another block from scratch.`